parsers/cxx/cxx_parser.c

```diff
@@ -1485,7 +1485,10 @@ void cxxParserAnalyzeOtherStatement(void)
 			// out its proper scope. Better avoid emitting this one.
 			CXX_DEBUG_PRINT("But it has been preceded by the 'friend' keyword: this is not a real prototype");
 		} else {
-			int iScopesPushed = cxxParserEmitFunctionTags(&oInfo,CXXTagKindPROTOTYPE,CXXEmitFunctionTagsPushScopes,NULL);
+			int piCorkQueueIndex;
+			int iScopesPushed = cxxParserEmitFunctionTags(&oInfo,CXXTagKindPROTOTYPE,CXXEmitFunctionTagsPushScopes, &piCorkQueueIndex);
+			if (piCorkQueueIndex != CORK_NIL)
+				cxxParserSetEndLineForTagInCorkQueue (piCorkQueueIndex, g_cxx.pToken->iLineNumber);
```

Comment:
Here we are assuming that g_cxx.pToken points to the last token of the chain.
This is currently true. But maybe we could use the cxxTokenChainLast() instead to remove this dependency.

Reply:
Thank you. Updated.

Reply:
Could you merge this pull request if  the change is o.k.?

Reply:
Ow, sure!